File: src/main/java/net/hectus/neobb/turn/default_game/flower/TPoppy.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.player.NeoInventory;
import net.hectus.neobb.player.NeoPlayer;
import org.bukkit.block.Block;

import java.util.List;

public class TPoppy extends FlowerTurn {
    public TPoppy(NeoPlayer player) { super(player); }
    public TPoppy(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 6;
    }

    @Override
    public void apply() {
        List<NeoPlayer> players = player.game.players();

        NeoInventory temp = players.getFirst().inventory;
        for (int i = 0; i < players.size() - 1; i++) {
            players.get(i).inventory = players.get(i + 1).inventory;
        }
        players.getLast().inventory = temp;

        players.forEach(p -> p.inventory.switchOwner(p));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/flower/TPoppy.kt
package net.hectus.neobb.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoInventory
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TPoppy(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 6

    override fun apply() {
        val players = player!!.game.players

        val temp: NeoInventory = players.first().inventory
        for (i in 0..<players.size - 1) {
            players[i].inventory = players[i + 1].inventory
        }
        players.last().inventory = temp

        players.forEach { it.inventory.switchOwner(it) }
    }
}
